tests/test_gameboard.py

```diff
@@ -120,3 +120,55 @@ def test_winner_exists_from_diagonals_on_right_corner(self):
+                    [None, None, None, None, None, None, None],
+                    [None, None, None, None, None, None, None],
+                    [None, None, None, None, None, None, None],
+                    [None, None, None, None, None, None, None]])
+        board_2 = GameBoard.from_matrix([
+                    [None, None, None, None, None, None, None],
+                    [None, None, None, None, None, None, None],
+                    [None, None, None, None, None, None, None],
+                    [None, None, None, None, None, None, None],
+                    [None, None, None, None, None, None, None],
+                    [None, None, None, None, None, None, None]])
+        self.assertTrue(board_1 == board_2)
+
+    def test_equality_of_loaded_boards(self):
+        board_1 = GameBoard.from_matrix([
+                    ['blue', 'blue', 'blue',  'red',   None,   None, None],
+                    ['red',   'red',  'red', 'blue', 'blue',   None, None],
+                    ['blue',  'red', 'blue',  'red', 'blue',   None, None],
+                    ['blue',  'red', 'blue',  'red',  'red',  'red', None],
+                    ['blue', 'blue',  'red', 'blue',  'red', 'blue', None],
+                    ['red',   'red',  'red', 'blue',  'red', 'blue', None]])
+        board_2 = GameBoard.from_matrix([
+                    ['blue', 'blue', 'blue',  'red',   None,   None, None],
+                    ['red',   'red',  'red', 'blue', 'blue',   None, None],
+                    ['blue',  'red', 'blue',  'red', 'blue',   None, None],
+                    ['blue',  'red', 'blue',  'red',  'red',  'red', None],
+                    ['blue', 'blue',  'red', 'blue',  'red', 'blue', None],
+                    ['red',   'red',  'red', 'blue',  'red', 'blue', None]])
+        self.assertTrue(board_1 == board_2)
+
+
+    def test_non_equality_of_very_similar_boards(self):
+        board_1 = GameBoard.from_matrix([
+                    ['blue', 'blue', 'blue',  'red',   None,   None, None],
+                    ['red',   'red',  'red', 'blue', 'blue',   None, None],
+                    ['blue',  'red', 'blue',  'red', 'blue',   None, None],
+                    ['blue',  'red', 'blue',  'red',  'red',  'red', None],
+                    ['blue', 'blue',  'red', 'blue',  'red', 'blue', None],
+                    ['red',   'red',  'red', 'blue',  'red', 'blue', 'red']])
+        board_2 = GameBoard.from_matrix([
+                    ['blue', 'blue', 'blue',  'red',   None,   None,  None],
+                    ['red',   'red',  'red', 'blue', 'blue',   None,  None],
+                    ['blue',  'red', 'blue',  'red', 'blue',   None,  None],
+                    ['blue',  'red', 'blue',  'red',  'red',  'red',  None],
+                    ['blue', 'blue',  'red', 'blue',  'red', 'blue',  None],
+                    ['red',   'red',  'red', 'blue',  'red', 'blue', 'blue']])
+        self.assertFalse(board_1 == board_2)
```

Comment:
Another interesting case to test would be that of having the actual same matrix (remember equality vs identity):

```
matrix = [...]
board_1 = GameBoard.from_matrix(matrix)
board_2 = GameBoard.from_matrix(matrix)
```